project.mml

```diff
@@ -278,6 +278,24 @@ Layer:
       type: shape
     properties:
       minzoom: 8
+  - id: springs
+    geometry: point
+    <<: *extents
+    Datasource:
+      <<: *osm2pgsql
+      table: |-
+        (SELECT
+             St_Centroid(way) AS way, "natural"
```

Comment:
I'm not sure, but I suspect that this code makes test rendering on z14+ very long. Instead of 2-3 s it takes like 10+ s every time and there's heavy I/O load, which means big database query most probably.

This is nasty performance bug. Could you look closer and fix it?

Reply:
I think dropping St_Centroid should give a similar result?

Also this might result in the spring icon being displayed outside of the spring polygons in case of moon-shaped springs - do we want that? Dropping the St_Centroid should solve that too.

Reply:
This is beyond my knowledge, but sounds reasonable. I don't know how should the code look like after dropping it, could you show the proper code that I could test?

Reply:
Simply like this (like below):

way, "natural"

Make sure to test it on spring areas.

Reply:
What looks wrong according to you? Note that the text in the second item is not the same as the text in the first item.

Reply:
OK, it's just a mess with different interior algorithm again - when compared with my own master it looks OK, drinkable water name+icon is just moved with this area:

![tnkhxidd](https://user-images.githubusercontent.com/5439713/39454644-d8d8d60a-4cdc-11e8-9029-e3a76ad9aebb.png)

Reply:
For [all 6 such areas in Poland](http://overpass-turbo.eu/s/ypf) the icon is placed the same as before, so it looks like this is not even needed. I believe `marker-placement: interior` in MSS file does the job.

I'm not sure if we should wait for @Penegal with this or just update this line somehow and merge it?

Reply:
BTW: I hope soon we will have exactly the same interior algorithm in Kosmtik and on the OMSF servers, we're really close:
- https://github.com/kosmtik/kosmtik/pull/283 (and then a new Kosmtik release with npm package update, so our Docker image could be updated to...)
- https://github.com/openstreetmap/chef/issues/155#issuecomment-385318881